packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts

```diff
@@ -433,6 +436,26 @@ export const useGridFocus = (
     }
   }, [apiRef]);
 
+  const handlePaginationModelChange = useEventcallback(() => {
+    const currentFocusedCell = gridFocusCellSelector(apiRef);
+    if (!currentFocusedCell) {
+      return;
+    }
+
+    const currentPage = getVisibleRows(apiRef, {
+      pagination: props.pagination,
+      paginationMode: props.paginationMode,
+    });
+
+    const rowIsInCurrentPage = currentPage.rows.find((row) => row.id === currentFocusedCell.id);
+    if (rowIsInCurrentPage) {
+      return;
+    }
+
+    const visibleColumns = gridVisibleColumnDefinitionsSelector(apiRef);
+    apiRef.current.setCellFocus(currentPage.rows[0].id, visibleColumns[0].field);
```

Comment:
This doesn't feel optimal since it's now harder to change the page multiple times - you'll have to tab into pagination after each page change.
What if we only change the tabIndex state instead?

https://github.com/mui/mui-x/assets/13808724/4a9e31db-44f7-43fb-b277-de3250cebd63

What do you think?

```diff
diff --git a/packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts b/packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts
index e455e11e0..bbc8323f8 100644
--- a/packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts
+++ b/packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts
@@ -453,7 +453,17 @@ export const useGridFocus = (
     }
 
     const visibleColumns = gridVisibleColumnDefinitionsSelector(apiRef);
-    apiRef.current.setCellFocus(currentPage.rows[0].id, visibleColumns[0].field);
+    apiRef.current.setState((state) => {
+      return {
+        ...state,
+        tabIndex: {
+          cell: { id: currentPage.rows[0].id, field: visibleColumns[0].field },
+          columnHeader: null,
+          columnHeaderFilter: null,
+          columnGroupHeader: null,
+        },
+      };
+    });
   });
 
   const focusApi: GridFocusApi = {
```